### DISCUSSION TITLE
Is HmIP-WKP supported?

### DISCUSSION THREAD

**Discussion by marcin-mazurek** [SukramJ/aiohomematic#479] [General]
Is HmIP-WKP (https://homematic-ip.com/sites/default/files/downloads/hmip-wkp-156424a0-datasheet.pdf) supported by the library? If so, how should I listen for keyboard input events?

**Comment by SukramJ** [comment]
What happens after you installed the component?
So basic support should be there like described [here](https://github.com/danielperna84/custom_homematic#device-support).
To support the project [device definitions](https://github.com/danielperna84/custom_homematic#homematicip_localexport_device_definition) are required, that could be delivered by you, if you own such device.

But what i could see in short from the [technical documentation](https://homematic-ip.com/sites/default/files/downloads/HmIP_Device_Documentation-1.40.10.pdf) page 13798 there are no key press events from the keyboard.

**Reply by marcin-mazurek** [reply]
I do not own the device currently - but if basic support is available I'll most likely buy it and get back to you.

**Comment by Baxxy13** [comment]
Actually there are differences between CCU3 an RaspberryMatic for the "PRESS_LOCK" and "PRESS_UNLOCK" - Events in Channel :1 to :16.
CCU3 had this, latest official RM not. 
Since todays RM-Nightly, RM has it too because it uses now the same HmIP-Server-Component like CCU3.

But it's not clear which Version is correct. Maybe the "PRESS_xxx" Events are gone in the next CCU3 Firmware Release.
See... [Ticket in RaspberryMatic (german)](https://github.com/jens-maus/RaspberryMatic/issues/1846)
